src/lj_record.c

```diff
@@ -1667,9 +1667,11 @@ static void check_call_unroll(jit_State *J, TraceNo lnk)
   } else {
     if (count > J->param[JIT_P_callunroll]) {
       if (lnk) {  /* Possible tail- or up-recursion. */
+        GCproto *pt = (GCproto *)(((char *)J->pc)- sizeof(GCproto));
+        lua_assert(bc_op(*J->pc) == BC_JFUNCF);
 	lj_trace_flush(J, lnk);  /* Flush trace that only returns. */
 	/* Set a small, pseudo-random hotcount for a quick retry of JFUNC*. */
-	hotcount_set(J2GG(J), J->pc+1, LJ_PRNG_BITS(J, 4));
+        hotcount_set_pt(J2GG(J), pt, J->pc, LJ_PRNG_BITS(J, 4));
```

Comment:
indentation is weird here.

Reply:
well it changed from a tab to spaces but the indent is the same if you set tabsize to 8 like LuaJIT uses. If there was .editorconfig in the repo github would display this better.